core/src/main/java/org/apache/calcite/sql/validate/MustFilterRequirements.java

```diff
@@ -37,9 +35,9 @@ public class MustFilterRequirements {
 
   ImmutableBitSet mustFilterFields;
   ImmutableBitSet mustFilterBypassFields;
-  protected Set<SqlQualified> remnantMustFilterFields;
+  protected ImmutableSet<SqlQualified> remnantMustFilterFields;
   public MustFilterRequirements(ImmutableBitSet mustFilterFields,
-      ImmutableBitSet mustFilterBypassFields, Set<SqlQualified> remnantMustFilterFields) {
+      ImmutableBitSet mustFilterBypassFields, ImmutableSet<SqlQualified> remnantMustFilterFields) {
```

Comment:
Parameters and fields should be declared as Set not ImmutableSet

(I don't like the rule, but it minimizes binary compatibility problems for people who have shaded Guava.)

You will often see the following pattern in constructors:
```
  this.remnantMustFilterFields = ImmutableSet.copyOf(remnantMustFilterFields);
```
which ensures immutability and non-null.

Reply:
Does this apply to the first two parameters as well?
I see I could use `ImmutableBitSet.fromBitSet()`

Reply:
answer: No because it is not guava